### PR TITLE
PROVES-31 Change indexed record data structures

### PR DESCRIPTION
**Pull request by kehh**
* Structure now:
```json
"ca_objects/entity_series_date": [
            {
              "-": "2005-2010",
              "-dtr": {
                "gte": "2005-01-01T00:00:00Z",
                "lte": "2010-12-31T23:59:59Z"
              },
              "content_id": 1
            }
          ]
```
* Remove duplicated code
* deprecate call to serialize
* Make index property key calculation DRYer
* Add logging for errors when parsing values
* Introduce suffixes as constants
* Change ElasticSearch dynamic templates
  * Remove temporary and duplicate dynamic templates
  * Change `*content_ids` to `content_id` as new structure expects that
* Add special handling for ListItem attributes
  * Use keyword rather than fulltext search
* Store content_id alongside actual value
* Make log level configurable via text value

### CODE REVIEW

File: app/lib/Plugins/SearchEngine/Elastic8/FieldTypes/Length.php [hunk 1]
```diff
@@ -39,15 +39,10 @@
...
-	public function __construct($table_name, $element_code) {
-		parent::__construct($table_name, $element_code);
-	}
-
 	public function getIndexingFragment($content, array $options): array {
-		if (is_array($content)) {
-			$content = serialize($content);
-		}
-		if ($content == '') {
+		$this->setDefaultSuffix(self::SUFFIX_DOUBLE);
```
Comment (bri-gaia) [inline app/lib/Plugins/SearchEngine/Elastic8/FieldTypes/Length.php:43]: Would it make more sense to set this as a class property or in a construct? to ensure the default value is set consistently.

File: app/lib/Plugins/SearchEngine/Elastic8.php [hunk 2]
```diff
@@ -273,10 +274,11 @@ private function addFragmentToUpdateContentBuffer(
 	protected function getClient(): Client {
 		if (!self::$client) {
 			$logger = new \Monolog\Logger('elasticsearch');
-			$log_level = Monolog\Logger::WARNING;
+			$log_level = self::LOG_LEVEL_DEFAULT;
 			if (defined('__CA_ELASTICSEARCH_LOG_LEVEL__')) {
 				$log_level = __CA_ELASTICSEARCH_LOG_LEVEL__;
 			}
+			$log_level = \Monolog\Logger::toMonologLevel($log_level);
```
Comment (bri-gaia) [inline app/lib/Plugins/SearchEngine/Elastic8.php:281]: This was a little confusing at first, maybe we could use a different variable name for one of these or refactor this to make it more apparent?
(at first glance it seemed we were setting `$log_level` three times in a row with the final overriding the first two.)

**Comment by bri-gaia** [comment]
LGTM

File: app/lib/Plugins/SearchEngine/Elastic8.php (hunk 2 above)
Comment (kehh) [reply app/lib/Plugins/SearchEngine/Elastic8.php:281]: Will send this through as an update.

**Pull request merged**